src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodParameter.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeCracker.CSharp.Design.InconsistentAccessibility
+{
+    public sealed class InconsistentAccessibilityInMethodParameter : InconsistentAccessibilityInfoProvider
+    {
+        public async Task<InconsistentAccessibilityInfo> GetInconsistentAccessibilityInfoAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var result = new InconsistentAccessibilityInfo();
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodThatRaisedError = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).DescendantNodesAndSelf().OfType<BaseMethodDeclarationSyntax>().FirstOrDefault();
+            if (methodThatRaisedError != null)
+            {
+                var parameterTypeFromMessage = ExtractParameterTypeFromErrorMessage(diagnostic.ToString());
+
+                var parameterTypeLastIdentifier = parameterTypeFromMessage;
+                var parameterTypeDotIndex = parameterTypeLastIdentifier.LastIndexOf('.');
+                if (parameterTypeDotIndex > 0)
+                {
+                    parameterTypeLastIdentifier = parameterTypeLastIdentifier.Substring(parameterTypeDotIndex + 1);
+                }
+
+                result.TypeToChangeAccessibility = FindTypeSyntaxFromParametersList(methodThatRaisedError.ParameterList.Parameters, parameterTypeLastIdentifier);
+                result.CodeActionMessage = $"Change parameter type '{parameterTypeFromMessage}' accessibility to be as accessible as method '{GetMethodIdentifierValueText(methodThatRaisedError)}'";
+                result.NewAccessibilityModifiers = CreateAccessibilityModifiersFromMethod(methodThatRaisedError);
+            }
+
+            return result;
+        }
+
+        private static string ExtractParameterTypeFromErrorMessage(string compilerErrorMessage)
+        {
+            const int parameterTypeStartShift = 52;
+
+            var parameterTypeNameStart = compilerErrorMessage.IndexOf(InconsistentAccessibilityCodeFixProvider.InconsistentAccessibilityInMethodParameterCompilerErrorNumber, StringComparison.Ordinal) + parameterTypeStartShift;
+            var parameterTypeNameLength = compilerErrorMessage.IndexOf('\'', parameterTypeNameStart) - parameterTypeNameStart;
+            return compilerErrorMessage.Substring(parameterTypeNameStart, parameterTypeNameLength);
+        }
```

Comment:
You should not extract the parameter this way. We will have people that have different cultures in VS and the message will appear in different positions.
Try to change it to something like this:

````csharp
System.Text.RegularExpressions.Regex.Match(
    Microsoft.CodeAnalysis.CSharp.CSharpDiagnosticFormatter.Instance.Format(
        diagnostic, System.Globalization.CultureInfo.InvariantCulture),
        diagnostic.Descriptor.MessageFormat.ToString(System.Globalization.CultureInfo.InvariantCulture).Replace("{0}", "(.*)").Replace("{1}", "(.*)")
).Groups[1].Value
````